src/api.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+const assert = require('assert');
+const brorand = require('brorand');
+const hashjs = require('hash.js');
+const elliptic = require('elliptic');
+const Ed25519 = elliptic.eddsa('ed25519');
+const Secp256k1 = elliptic.ec('secp256k1');
+const addressCodec = require('ripple-address-codec');
+const deriveSecret = require('./secp256k1').deriveSecret;
+const utils = require('./utils');
+const hexToBytes = utils.hexToBytes;
+const bytesToHex = utils.bytesToHex;
+
+function generateSeed(options = {}) {
+  assert(!options.entropy || options.entropy.length >= 16, 'entropy too short');
+  const entropy = options.entropy ? options.entropy.slice(0, 16) : brorand(16);
+  return addressCodec.encodeSeed(entropy, options.algorithm || 'secp256k1');
+}
+
+const secp256k1 = {
+  deriveKeypair: function(entropy, options) {
+    const prefix = '00';
+    const privateKey = prefix + deriveSecret(entropy, options)
+      .toString(16, 64).toUpperCase();
+    const publicKey = bytesToHex(
+      Secp256k1.keyFromPrivate(privateKey).getPublic().encodeCompressed());
+    return {privateKey, publicKey};
+  },
+  sign: function(message, privateKey) {
+    const digest = hashjs.sha512().update(message).digest().slice(0, 32);
+    const key = Secp256k1.keyFromPrivate(hexToBytes(privateKey));
+    return bytesToHex(key.sign(digest, {canonical: true}).toDER());
+  },
+  verify: function(signature, message, publicKey) {
+    const key = Secp256k1.keyFromPublic(hexToBytes(publicKey));
+    const digest = hashjs.sha512().update(message).digest().slice(0, 32);
+    return key.verify(digest, signature);
+  }
+};
+
+const ed25519 = {
+  deriveKeypair: function(entropy) {
+    const prefix = 'ED';
+    const rawPrivateKey = new utils.Sha512().add(entropy).first256();
```

Comment:
`new utils.Sha512().add(entropy).first256();`

We should add a helper function for this huh, as it's so common.  A static method, using the existing class as namespace?  `Sha512.first256(entropy)` or something ?